js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSRunner.scala

```diff
@@ -38,12 +38,25 @@ trait AsyncJSRunner {
 
   /** Await completion of the started Run for the duration specified
    *  by [[atMost]]. Strictly equivalent to:
-   * 
+   *
    *  {{{
    *  Await.result(future, atMost)
    *  }}}
-   * 
+   *
    */
   final def await(atMost: Duration): Unit = Await.result(future, atMost)
 
+  /** Await completion of the started Run for the duration specified by
+   *  [[atMost]], or force it to stop if the timeout expires.
+   */
+  final def awaitOrStop(atMost: Duration): Unit = {
+    try {
+      await(atMost)
+    } catch {
+      case th: Throwable =>
```

Comment:
Ok. Agreed. But then you have to change the documentation of `stop` explicitly to say that it **must** be a no-op if the runner is completed (and verify that this is actually the case).

Further, consider using a finally block in this case.